Client/Frontend/Browser/HomePageHelper.swift

```diff
@@ -50,4 +50,27 @@ class HomePageHelper {
         tab.loadRequest(NSURLRequest(URL: url))
     }
 
+    func openHomePage(inTab tab: Tab, withNavigationController navigationController: UINavigationController?) {
+        if isHomePageAvailable {
+            openHomePage(tab)
+        } else {
+            setHomePage(toTab: tab, withNavigationController: navigationController)
+        }
+    }
+
+    func setHomePage(toTab tab: Tab, withNavigationController navigationController: UINavigationController?) {
+        let alertController = UIAlertController(
+            title: Strings.SetHomePageDialogTitle,
+            message: Strings.SetHomePageDialogMessage,
+            preferredStyle: UIAlertControllerStyle.Alert)
+        alertController.addAction(
+            UIAlertAction(title: Strings.SetHomePageDialogNo, style: .Cancel) { (action) in
+                // Do nothing.
+            })
+        alertController.addAction(
+            UIAlertAction(title: Strings.SetHomePageDialogYes, style: .Default) { (action) in
+                self.currentURL = tab.url
```

Comment:
Yes, there is a retain chain `UIAlertController` -> `Action` -> `self` (`HomePageHelper`). However, there is no retain path from the helper to `UIAlertController`.

You can see this by seeing 

```swift
{ [weak self] (action) in
    guard let _ = self else {
        return print("Homepage not set")
    }
    self?.currentURL = tab.url
})
```
You can also prove this by adding a `deinit` which logs to the console.

cc @sleroux 

For future Google searches, using `unowned` to explicit break any cycle (had it existed), and to document what else I found:
```swift
{ [unowned self] (action) in
    self.currentURL = tab.url
})
```

Cases a runtime crash:
```
attempted to retain deallocated object
```